src/main/java/appeng/api/storage/IMEInventory.java

```diff
@@ -50,13 +50,26 @@ public interface IMEInventory<StackType extends IAEStack> {
     StackType extractItems(StackType request, Actionable mode, BaseActionSource src);
 
     /**
-     * request a full report of all available items, storage.
+     * Request a full report of all available items, storage.
      *
      * @param out the IItemList the results will be written too
      * @return returns same list that was passed in, is passed out
      */
     IItemList<StackType> getAvailableItems(IItemList<StackType> out);
 
+    /**
+     * Request a report of how many of a single item type are available in storage. It falls back to
+     * {@link IMEInventory#getAvailableItems} if a more optimized implementation is not present.
+     *
+     * @param request The item type to search for, it does not get modified
+     * @return A new stack with the stack size set to the count of items in storage, or null if none are present
+     */
+    @SuppressWarnings("unchecked") // changing the generic StackType to be correct here is too much of a breaking API
+                                   // change
+    default StackType getAvailableItem(StackType request) {
```

Comment:
nit: This should probably be annotated with `@Nullable` or equivalent

Reply:
I added NonNull to the request, Nullable is the default in ae2 (and most mc mods) as there is no package-level override.